libs/core/algorithms/tests/regressions/for_loop_test.cpp

```diff
@@ -0,0 +1,32 @@
+//  Copyright (c) 2022 Deepak Suresh
+//
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+
+#include <hpx/local/algorithm.hpp>
+#include <hpx/local/execution.hpp>
+#include <hpx/local/init.hpp>
+#include <hpx/modules/testing.hpp>
+
+#include "hpx/hpx.hpp"
+#include "hpx/hpx_init.hpp"
+
+int hpx_main(int argc, char *argv[])
+{
+    int start = 3;
+    int end = 7;
```

Comment:
Correct me if I am wrong, but shouldn't `start` be greater than `end` to test the corresponding issue? I am probably missing something here?

Reply:
@gonidelis even I thought that, 
but we write these tests for checking correctness of the program right ? so, I changed it to a valid `for_loop` arguments. Should I change it.

Reply:
> @gonidelis even I thought that, but we write these tests for checking correctness of the program right ? so, I changed it to a valid `for_loop` arguments. Should I change it.

As said before, a unit test is designed to fail without the patch but succeeds after applying it. Please change your code.